packages/permission-log-controller/tests/PermissionLogController.test.ts

```diff
@@ -0,0 +1,759 @@
+      logMiddleware: JsonRpcMiddleware<JsonRpcParams, Json>;
+
+    beforeEach(() => {
+      permLog = initPermLog();
+      logMiddleware = initMiddleware(permLog);
+      initClock();
+    });
+
+    afterEach(() => {
+      tearDownClock();
+    });
+
+    it('only updates history on responses', () => {
+      const req = RPC_REQUESTS.requestPermission(
+        SUBJECTS.a.origin,
+        PERM_NAMES.test_method,
+      );
+      const res = {
+        ...PendingJsonRpcResponseStruct.TYPE,
+        result: [PERMS.granted.test_method()],
+      };
+
+      // noop => no response
+      logMiddleware(req, res, noop, noop);
+
+      expect(permLog.getHistory()).toStrictEqual({});
+
+      // response => records granted permissions
+      logMiddleware(req, res, mockNext, noop);
+
+      const permHistory = permLog.getHistory();
+      expect(Object.keys(permHistory)).toHaveLength(1);
+      expect(permHistory[SUBJECTS.a.origin]).toBeDefined();
+    });
+
+    it('ignores malformed permissions requests', () => {
+      const req = RPC_REQUESTS.requestPermission(
+        SUBJECTS.a.origin,
+        PERM_NAMES.test_method,
+      );
+      const res = {
+        ...PendingJsonRpcResponseStruct.TYPE,
+        result: [PERMS.granted.test_method()],
+      };
+
+      // no params => no response
+      logMiddleware(
+        {
+          ...req,
+          params: undefined,
+        },
+        res,
+        mockNext,
+        noop,
+      );
+
+      expect(permLog.getHistory()).toStrictEqual({});
+    });
+
+    it('records and updates account history as expected', async () => {
+      const req = RPC_REQUESTS.requestPermission(
+        SUBJECTS.a.origin,
+        PERM_NAMES.eth_accounts,
+      );
+      const res = {
+        ...PendingJsonRpcResponseStruct.TYPE,
+        result: [PERMS.granted.eth_accounts(ACCOUNTS.a.permitted)],
+      };
+
+      logMiddleware(req, res, mockNext, noop);
+
+      expect(permLog.getHistory()).toStrictEqual(EXPECTED_HISTORIES.case1[0]);
+
+      // mock permission requested again, with another approved account
+
+      clock.tick(1);
+
+      res.result = [PERMS.granted.eth_accounts([ACCOUNTS.a.permitted[0]])];
+
+      logMiddleware(req, res, mockNext, noop);
+
+      expect(permLog.getHistory()).toStrictEqual(EXPECTED_HISTORIES.case1[1]);
+    });
+
+    it('handles eth_accounts response without caveats', async () => {
+      const req = RPC_REQUESTS.requestPermission(
+        SUBJECTS.a.origin,
+        PERM_NAMES.eth_accounts,
+      );
+      const res: PendingJsonRpcResponse<Permission[]> = {
+        ...PendingJsonRpcResponseStruct.TYPE,
+        result: [PERMS.granted.eth_accounts(ACCOUNTS.a.permitted)],
+      };
+      delete res.result?.[0].caveats;
+
+      logMiddleware(req, res, mockNext, noop);
+
+      expect(permLog.getHistory()).toStrictEqual(EXPECTED_HISTORIES.case2[0]);
+    });
+
+    // wallet_requestPermissions returns all permissions approved for the
```

Comment:
It looks like we removed the test "handles extra caveats for eth_accounts". Was there a reason for this?

Reply:
Yes I did remove this case as it was a case that's not relevant for typescript we were adding some invalid caveats (which is not possible with typing) and we were waiting for them not to be available on the permissionHistory